src/holon/rule_engine/repositories/repository_base.py

```diff
@@ -1,30 +1,46 @@
 from __future__ import annotations
-from typing import Type
+
+from polymorphic import utils
+import copy
+from typing import Type, TypeVar
+from django.db.models.base import Model as DjangoModel
+from polymorphic.models import PolymorphicModel
 
 from holon.models.filter.attribute_filter_comparator import AttributeFilterComparator
 
 
 class RepositoryBaseClass:
     """Repository containing all actors in memory"""
 
-    objects: list[object] = []
+    objects: list[PolymorphicModel] = []
 
     def dict(self):
         return {obj.id: obj for obj in self.objects}
 
-    # TODO ERIK
     def clone(self) -> RepositoryBaseClass:
         """Clone the object"""
+        # TODO: seems better to set the objects via the constructor in the future
```

Comment:
Ik ben het eens met deze comment, kunnen we dit gelijk doen?
Misschien iets als dit:
```
cloned_objects = copy.copy(self.objects)
return self.__class__(objects=cloned_objects)
```
En dan een constructor die een optionele objects aanneemt (`objects: list[PolymorphicModel = None` in `__init__` params). 
Wat vind je hiervan?

Reply:
Ik zou zeggen dat de constructor altijd objects neemt. En dat er daarnaast een "named constructor" is. 

Zoiets als hieronder, met actor_dict als voorbeeld. 

``` python
class BaseRepository:
    def __init__(self, objects: list):
        self.objects = objects

class ActorRepository(BaseRepository):
    actor_dict: dict

    def __init__(objects: list):
        super(objects)
        self.actor_dict = { comprehension }

    @classmethod
    def from_scenario(cls, scenario: Scenario):
        return cls(
            Actor.objects.filter(payload=scenario).get_real_instances()
        )
```

Reply:
Ik maak een losse PR ok? laten we het klein houden met korte doorlooptijd

Reply:
Toch in deze PR. Done.